core/src/main/java/com/datastrato/gravitino/proto/UserEntitySerDe.java

```diff
@@ -30,7 +31,8 @@ public UserEntity deserialize(User user) {
         UserEntity.builder()
             .withId(user.getId())
             .withName(user.getName())
-            .withAuditInfo(new AuditInfoSerDe().deserialize(user.getAuditInfo()));
+            .withAuditInfo(new AuditInfoSerDe().deserialize(user.getAuditInfo()))
+            .withRoles(Lists.newArrayList());
```

Comment:
passed an empty role list when serialized, but return null when deserialized，this cause ut failure

and in relation backend，it also return emtpy list instead of null，we need to keep consistent